docs/ensembl-help/using-ensembl/getting-ensembl-data/tools-and-apis/refget/about-refget.md

```diff
@@ -0,0 +1,375 @@
+
+```json
+{
+  "data": {
+    "genomes": [
+      {
+        "genome_id": "a7335667-93e7-11ec-a39d-005056b38ce3"
+      }
+    ]
+  }
+}
+```
+
+We can use the following GraphQL query to retrieve all known transcripts for the human BRCA2 gene using the previously retrieved `genome_id`.
+
+```json
+query {
+  genes(
+    by_symbol: {
+      symbol: "BRCA2",
+      genome_id: "a7335667-93e7-11ec-a39d-005056b38ce3"
+    }
+  ) {
+    stable_id
+    symbol
+    transcripts {
+      stable_id
+      symbol
+      metadata {
+        canonical {
+          value
+        }
+      }
+      product_generating_contexts {
+        product_type
+        cdna {
+          sequence {
+            checksum
+          }
+        }
+        cds {
+          sequence {
+            checksum
+          }
+        }
+        product {
+          stable_id
+          sequence {
+            checksum
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+Each sequence product (cDNA, CDS and protein) refer to a checksum. The value of this field should be used verbatim with the refget service.
+
+```json
+{
+  "data": {
+    "genes": [
+      {
+        "stable_id": "ENSG00000139618.18",
+        "symbol": "BRCA2",
+        "transcripts": [
+          {
+            "stable_id": "ENST00000380152.8",
+            "symbol": "BRCA2-201",
+            "metadata": {
+              "canonical": {
+                "value": true
+              }
+            },
+            "product_generating_contexts": [
+              {
+                "product_type": "Protein",
+                "cdna": {
+                  "sequence": {
+                    "checksum": "2f8c687a656debd4c8b1b4d47747ba35"
+                  }
+                },
+                "cds": {
+                  "sequence": {
+                    "checksum": "c6b5e532b5aef696e987ca55a538ae9e"
+                  }
+                },
+                "product": {
+                  "stable_id": "ENSP00000369497.3",
+                  "sequence": {
+                    "checksum": "92addb948c6c652abc1dcecca05f26c0"
+                  }
+                }
+              }
+            ]
+          },
+          ....
+        ]
+      }
+    ]
+  }
+}
+```
+
+Each sequence can now be retrieved with the following URLs.
+
+- cDNA: <https://beta.ensembl.org/data/refget/sequence/2f8c687a656debd4c8b1b4d47747ba35>
+- CDS: <https://beta.ensembl.org/data/refget/sequence/c6b5e532b5aef696e987ca55a538ae9e>
+- Protein: <https://beta.ensembl.org/data/refget/sequence/92addb948c6c652abc1dcecca05f26c0>
+
+Metadata concerning each of the records can be retreived by adding a `/metadata` to each URL.
+
+- cDNA: <https://beta.ensembl.org/data/refget/sequence/2f8c687a656debd4c8b1b4d47747ba35/metadata>
+- CDS: <https://beta.ensembl.org/data/refget/sequence/c6b5e532b5aef696e987ca55a538ae9e/metadata>
+- Protein: <https://beta.ensembl.org/data/refget/sequence/92addb948c6c652abc1dcecca05f26c0/metadata>
```

Comment:
I think you are missing in your example the very important use case of fetching genomic sequences.

Genomic sequences of features located in the genomic context (genes, transcripts, exons) do not have checksums of their own, but rather, require the user to define a slice of the sequence of the region on which they are located.

Thus, to get e.g. the sequence of a transcript, one would:

- include in the graphql query `transcript->slice->region->sequence->checksum`, which will return the checksum of the whole top-level region (e.g. chromosome); as well as `transcript->slice->location->start|end`, which will provide the genomic coordinates of the transcript (in Ensembl 1-based coordinates); and then send a request to refget for that region specifying the start and the end coordinates (in refget 0-based coordinates).

This will be a pretty involved example :-) It might require a separate markdown document, just for it.